modules/pre-commit.nix

```diff
@@ -381,26 +381,28 @@ in
           elif ! ${git}/bin/git rev-parse --git-dir &> /dev/null; then
             echo 1>&2 "WARNING: pre-commit-hooks.nix: .git not found; skipping installation."
           else
+            GIT_WC=`${git}/bin/git rev-parse --git-dir`/..
```

Comment:
This will put the file in `.git/worktrees` regardless of which worktree you're in.
If you're going the `--git-dir` route, you need to pit it inside the git dir. I think doing that was already suggested somewhere.

Reply:
Ah, so basically we want:

```sh
GIT_WC=`${git}/bin/git rev-parse --git-dir`
```

[Then](https://stackoverflow.com/a/75780328/55246), the YAML file would be put under `.git/worktrees/foo` - and thus we can obviate having to gitignore it.

Reply:
Different worktrees may use different revisions of the hooks, leading to a branch to be reformatted with the wrong logic.

Reply:
Besides, `.git` can also exist completely outside of the (only) working tree. So the `/..` appending is wrong.

So, if I understand you correctly, `GIT_WC=`${git}/bin/git rev-parse --git-dir`` is where we should put the YAML. That is guaranteed to return a path unique to that worktree.

Reply:
I don't think `pre-commit --config` is clever enough to work with the right `--git-dir` though. It appears as a relative path that's hardcoded in the hook script. And there's only one hook script.

It doesn't have shell injection like `pre-commit install --config='.git/pre-commit-hooks.yaml'`, and it's not clever about `.git/`.

Reply:
Maybe just go with `git rev-parse --show-toplevel`?
That seems to have been your original intent and seems to work fine.

Reply:
Alright, done.